baballonia: init at 1.1.0.8

ZenIsBestWolf opened the pull request:
Add `baballonia`
Add `babble` to the license list
Add `zenisbestwolf` to the maintainer list

This package would not exist without the original work done by @ShyAssassin 
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ZenIsBestWolf committed:
maintainers: add zenisbestwolf

Scrumplex commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,104 @@
+  fetchFromGitHub,
+}:
+let
+  internal = builtins.fetchurl {
```
You should use `pkgs.fetchurl` instead, to avoid the pure eval error we can see in the CI. Make sure to add `fetchurl` to the args list above

```suggestion
  internal = fetchurl {
```

ToasterUwU reviewed (changes requested):
Collapse the commit for fixing the fetch stuff into the init commit for baballonia. 
Extra commits are unwanted, one commit per package or other thing is the norm.

ToasterUwU commented:
Also, has @ShyAssassin agreed to being added as a maintainer? Being a maintainer has the expectation attached to it that they will respond to issues regarding the package and actually maintain it. So if this entry is meant for giving credit only, this is not the way to do it. Of course, if @ShyAssassin knows you added them to the list and is ok with it, thats totally fine.

ZenIsBestWolf commented:
> Also, has @ShyAssassin agreed to being added as a maintainer? Being a maintainer has the expectation attached to it that they will respond to issues regarding the package and actually maintain it. So if this entry is meant for giving credit only, this is not the way to do it. Of course, if @ShyAssassin knows you added them to the list and is ok with it, thats totally fine.

ShyAssassin agreed on Discord to be a maintainer when the package was being submitted to `nixpkgs-xr`, but not `nixpkgs` properly. It may be worthwhile to allow for a response here just to confirm.

ZenIsBestWolf commented:
Also, collapsed that commit. Thank you.

ShyAssassin commented:
>Also, has @ShyAssassin agreed to being added as a maintainer? 
>> ShyAssassin agreed on Discord to be a maintainer when the package was being submitted to nixpkgs-xr, but not nixpkgs properly. It may be worthwhile to allow for a response here just to confirm.

I did agree to maintaining it in private messages with @ZenIsBestWolf but just for clarification, I am more than happy to be one of the maintainers of the `baballonia` package.

ToasterUwU commented on pkgs/by-name/op/opencvsharp/package.nix:
With this added context, i agree with this

coolGi69 commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,105 @@
+    platforms = lib.platforms.linux;
+    homepage = "https://github.com/Project-Babble/Baballonia";
+    description = "Free and open source eye and face tracking for social VR";
+    license = {
```
Shouldn't this licence be included in the [`lib/licences.nix`](https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix) file incase other projects need it (ie, EyeTrackVR).

ZenIsBestWolf replied:
I'm perfectly happy to add it to the license file, I just didn't think anything else utilized the Project Babble license.

coolGi69 commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,105 @@
+  dotnetRuntime = dotnet.runtime;
+  projectFile = "src/Baballonia.Desktop/Baballonia.Desktop.csproj";
+
+  runtimeDeps = baballoniaPrograms;
```
Currently testing locally, and this doesn't seem to be necessary. (Though, someone else should probably test this as well incase I made a mistake)\
So the `baballoniaPrograms` var could just be put into the build inputs.

coolGi69 replied:
Ah alright, guess I didn't test thoroughly enough.

ZenIsBestWolf replied:
Quick testing on my side, all that was needed was `udev`, so I have made this change. Thank you!

coolGi69 reviewed (changes requested):
Did a bit more of an in-depth review on this to remove some unneeded stuff from the package. Sry for missing these on the `nixpkgs-xr` review 😅

ZenIsBestWolf commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,105 @@
+  pname = "baballonia";
+
+  patches = [
+    # Disable automatic updates
```
This was required by the formatter.

ZenIsBestWolf commented:
Thank you all again for the feedback and requested changes, this is my first time packaging any software whatsoever, and it is incredibly exciting to be able to contribute back to the greater community.

ShyAssassin commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,105 @@
+  ];
+
+  postUnpack = ''
+    cp ${internal} $sourceRoot/src/Baballonia.Desktop/_internal.zip
```
Personally i dont really see the need to link instead of copy here since this is just for building and the final package *shouldnt* include this zip folder so it wouldnt contribute to size, unless i misunderstand how `postUnpack` works.

coolGi69 replied:
I was thinking it'd probs be better to avoid a copy, isn't needed.

coolGi69 commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,105 @@
+let
+  internal = fetchurl {
+    # This URL is weird but this is the primary source
+    url = "http://217.154.52.44:7771/builds/trainer/1.0.0.0.zip";
```
Isn't it still used in the [fetch_internal.ps1](https://github.com/Project-Babble/Baballonia/blob/c848ff5c994f36155f66e2d8c17a8218351881f9/src/Baballonia.Desktop/fetch_internal.ps1#L2)? The file doesn't seem to have updated...

ShyAssassin replied:
We are also targeting the `1.1.0.8` release tag which seems to use it, but if it isnt used in later version it should be removed when we bump package versions.

ZenIsBestWolf commented:
I've implemented the major feedback requested from reviewers, so this should be good to go (or at least for another round of reviews).

The only thing I may want to change/ask to clarify: I removed the `meta` tag from the `opencvsharp` definition now that it is not a dedicated package. Was this appropriate to do, or should I re-add it?

coolGi69 commented on lib/licenses.nix:
```diff
@@ -167,6 +167,13 @@ lib.mapAttrs mkLicense (
       fullName = "Apache License 2.0";
     };
 
+    babble = {
```
I'm not 100% sure on this, but should it be `babble1` and have the URL point to a specific commit? Just incase it ever updates.

Again, not 100% sure about this as some licenses include the version number, while others don't...

coolGi69 reviewed (approved):
LGTM! Hopefully this can get merged soon if there aren't any other issues!

Scrumplex commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,122 @@
+  internal = fetchurl {
+    # This URL is weird but this is the primary source
+    url = "http://217.154.52.44:7771/builds/trainer/1.0.0.0.zip";
+    sha256 = "sha256:0cfc1r1nwcrkihmi9xn4higybyawy465qa6kpls2bjh9wbl5ys82";
```
nit:

Using `hash` is preferred over `sha256` for code style consistency. Also, SRI-hashes are more prevalent in the ecosystem, so perhaps it might be a good idea to do that here (`sha256-<base64>` instead of your current hash)

Scrumplex commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,122 @@
+    xorg.libSM
+    xorg.libICE
+    opencvsharp
+  ];
```
unused?

Scrumplex commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,122 @@
+    mv $out/lib/baballonia/Baballonia.IPCameraCapture.dll $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.IPCameraCapture.pdb $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.SerialCameraCapture.dll $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.SerialCameraCapture.pdb $out/lib/baballonia/Modules/
```
Maybe we would want to use a wildcard here to avoid unexpected failures in case new libraries get added in the future.


```suggestion
    mv $out/lib/baballonia/Baballonia.*.{dll,pdb} $out/lib/baballonia/Modules/
```

coolGi69 replied:
@Scrumplex I just wanted to say I did try this in the past, but it moved more things that are necessary. It may be causing https://github.com/NixOS/nixpkgs/pull/459868#issuecomment-3538681064 somehow? 

Though, I'm not sure on how else to improve the situation.
Maybe at least using an array of the names of the things that need to be moved, and just iterating over that?

ZenIsBestWolf committed:
licenses: add babble1

ZenIsBestWolf commented:
@Scrumplex The main executable should still be `baballonia`, not `Baballonia.Desktop`. I am not able to reproduce this error when using the main program, how are you executing this?

As for the `.so` files found in `$out/bin`, that's odd, I'm not sure why they are placed there. What step should I go about moving them such that the application can still launch?

ZenIsBestWolf committed:
baballonia: init at 1.1.0.8

Co-authored-by: ShyAssassin <ShyAssassin@assassin.dev>

coolGi69 commented on pkgs/by-name/ba/baballonia/package.nix:
```diff
@@ -0,0 +1,105 @@
+    platforms = lib.platforms.linux;
+    homepage = "https://github.com/Project-Babble/Baballonia";
+    description = "Free and open source eye and face tracking for social VR";
+    license = lib.licenses.babble1;
```
As I said in https://github.com/NixOS/nixpkgs/pull/459868#discussion_r2507585230, [EyeTrackVR](https://github.com/EyeTrackVR/EyeTrackVR) also makes use of this licence incase we ever wanna package that. I'm sure there are others (including stuff made by Project Babble as TosterUwU said).

ToasterUwU reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

coolGi69 commented:
On another note, should a `.desktop` file be made as this is a GUI app?

Naraenda commented:
The recent changes in the `.csproj`-files made the post-fixup much easier. Also the trainer/callibration tool should be fetched from the other repositories. I'm currently just fetching latest release (like what baballonia does in CI). [You can see checkout the out-of-tree package I wrote for it.](https://github.com/Naraenda/nix-config/blob/develop/pkgs/baballonia-git/default.nix)

ToasterUwU commented:
There is a v1.1.0.9 out now, it has significant changes, so might as well update this package before merging instead of merging a old version just to edit it instantly after again
